### PR TITLE
Add decoded ILAttribute representation

### PR DESCRIPTION
**Pull request by auduchinok**
Allows passing decoded custom attributes when reading IL metadata.

One thing to consider is equality for this type as there're now two cases with different representations and structural equality cannot be used anymore. Are these attributes objects ever checked for equality?

### CODE REVIEW

File: src/absil/il.fsi [hunk 1]
```diff
@@ -752,12 +752,15 @@ type ILAttribElem =
...
-/// to ILAttribElem's as best as possible.  
+/// Custom attributes. See 'decodeILAttribData' for a helper to parse the byte[] to ILAttribElem's as best as possible.
 type ILAttribute =
-    { Method: ILMethodSpec  
-      Data: byte[] 
-      Elements: ILAttribElem list}
+    | Encoded of method: ILMethodSpec * data: byte[] * elements: ILAttribElem list
+    | Decoded of method: ILMethodSpec * fixedArgs: ILAttribElem list * namedArgs: ILAttributeNamedArg list
+
+    member Method: ILMethodSpec
```
Comment (dsyme) [inline src/absil/il.fsi]: `///` docs please

File: src/absil/il.fs [hunk 2]
```diff
@@ -947,11 +947,25 @@ type ILAttribElem =
...
 type ILAttributeNamedArg =  (string * ILType * bool * ILAttribElem)
 
-[<StructuralEquality; StructuralComparison; StructuredFormatDisplay("{DebugText}")>]
-type ILAttribute = 
-    { Method: ILMethodSpec
-      Data: byte[] 
-      Elements: ILAttribElem list }
+[<RequireQualifiedAccess; StructuralEquality; StructuralComparison; StructuredFormatDisplay("{DebugText}")>]
+type ILAttribute =
+    | Encoded of method: ILMethodSpec * data: byte[] * elements: ILAttribElem list
```
Comment (dsyme) [inline src/absil/il.fs]: `///` docs please

**Comment by dsyme** [comment]
@auduchinok Now, we never compare for equality AFAIK.

File: src/absil/il.fs (hunk 2 above)
Comment (auduchinok) [reply src/absil/il.fs]: Sure, will add.

**Pull request merged**